.github/workflows/build-gpu.yml

```diff
@@ -0,0 +1,46 @@
+name: Build GPU
+
+on:
+  push:
+  pull_request:
```

Comment:
Do we want to run this on every single PR and all push events to all branches? That's probably unnecessary and potentially $$$.

Reply:
> Do we want to run this on every single PR and all push events to all branches? That's probably unnecessary and potentially $$$.

It isn't necessary at all. We can just run once daily as well.

Here is a rough estimate of the costs:

## GPU & Instance Pricing
`n1-standard` = $0.04749975 per hour
`NVIDIA® Tesla® T4` = $0.35 per
Total = $ 0.3975 per hour ~ 0.006625 per minute

Source: 
- https://cloud.google.com/compute/gpus-pricing
- https://cloud.google.com/compute/vm-instance-pricing#n1_predefined

##  GPU Tests Run cost (per push two builds)
Each run is about 12 min x 2 (Python 3.7, 3.8) = 24 min x 0.006625 = **$0.159**

- 1 Builds a day = $0.159 x 1 x 30 ~ **$4.77 per month**
- 10 Builds a day = $0.159 x 10 x 30 ~ **$47.7 per month**

Reply:
I don't have the build stats on hand, but the current config is definitely an overkill. A lagom setup could be:

```yaml
on:
  workflow_dispatch:
  push:
    branches:
      - main
  pull_request:
```

Which is PR changes + push to `main` + manual trigger. And if we want to save $, we could remove the PR trigger. But I will let other comment how much we are willing to spend here. ping @hammer

Reply:
I'm fine with a minimal setup for now so removing the PR trigger and just firing on pushes to `main` and manually works for me.